tests/kubernetes-plugins/full.bats

```diff
@@ -64,38 +76,82 @@ function deploy_fluent_bit() {
     fi
 }
 
-@test "test fluent-bit adds k8s labels to records" {
-    deploy_fluent_bit
 
+function create_test_pod() {
+    TEST_POD_NAME="$1"
     # The hello-world-1 container MUST be on the same node as the fluentbit worker, so we use a nodeSelector to specify the same node name
     run kubectl get pods $FLUENTBIT_POD_NAME -n $TEST_NAMESPACE -o jsonpath='{.spec.nodeName}'
     assert_success
     refute_output ""
     node_name=$output
 
-    kubectl run -n $TEST_NAMESPACE hello-world-1 --image=docker.io/library/alpine:latest -l "this_is_a_test_label=true" \
+    kubectl run -n $TEST_NAMESPACE $TEST_POD_NAME --image=docker.io/library/alpine:latest -l "this_is_a_test_label=true" \
         --overrides="{\"apiVersion\":\"v1\",\"spec\":{\"nodeSelector\":{\"kubernetes.io/hostname\":\"$node_name\"}}}" \
         --command -- sh -c 'while true; do echo "hello world"; sleep 1; done'
 
     try "at most 15 times every 5s " \
-        "to find 1 pods named 'hello-world-1' " \
+        "to find 1 pods named '$TEST_POD_NAME' " \
         "with 'status' being 'Running'"
-
+    
     # We are sleeping here specifically for the Fluent-Bit's tail input's
     # configured Refresh_Interval to have enough time to detect the new pod's log file
     # and to have processed part of it.
     # A future improvement instead of sleep could use fluentbit's metrics endpoints
-    # to know the tail lugin has processed records
+    # to know the tail plugin has processed records
     sleep 10
+}
+
+function check_fluent_output() {
+    HAS_POD_LABELS=$1
+    HAS_NAMESPACE_LABELS=$2
 
-    run kubectl logs -l "app.kubernetes.io/name=fluent-bit" -n "$TEST_NAMESPACE"
+    run kubectl logs -l "app.kubernetes.io/name=fluent-bit" -n "$TEST_NAMESPACE" --tail=1
     assert_success
     refute_output ""
-    match1='kubernetes":{"pod_name":"hello-world-1","namespace_name":'
-    match1=${match1}\"${TEST_NAMESPACE}\"
+    
+    # Check pod label matches
+    match1="kubernetes\":{\"pod_name\":\"${TEST_POD_NAME}\",\"namespace_name\":\"${TEST_NAMESPACE}\""
     match2='"labels":{"this_is_a_test_label":"true"}'
+    if [ "$HAS_POD_LABELS" = true ]; then
+        assert_output --partial $match1
+        assert_output --partial $match2
+    else
+        refute_output --partial $match1
+        refute_output --partial $match2
+    fi
 
-    assert_output --partial $match1
-    assert_output --partial $match2
+    # Check namespace label matches
+    match3="\"kubernetes_namespace\":{\"name\":\"${TEST_NAMESPACE}\",\"labels\":{\""
+    match4='"this_is_a_namespace_label":"true"'
+    if [ "$HAS_NAMESPACE_LABELS" = true ]; then
+        assert_output --partial $match3
+        assert_output --partial $match4
+    else
+        refute_output --partial $match3
+        refute_output --partial $match4
+    fi
+}
+
+@test "test fluent-bit adds kubernetes pod labels to records" {
+    set_fluent_bit_pod_name
+    create_test_pod "k8s-pod-label-tester"
+    check_fluent_output true false
```

Comment:
These are bit esoteric, let's add a wrapper function with a proper name for the combinations that just forwards, e.g. `check_fluent_output_has_namespace_labels` or something like that maybe?
It's easier to read and review the intent then to ensure we don't accidentally add the wrong combination of true/false.